src/transaction/TransactionController.ts

```diff
@@ -1282,19 +1313,22 @@ export class TransactionController extends BaseController<
           transactionHash,
         ]);
 
+        if (meta.pollingAttempts === undefined) {
+          meta.pollingAttempts = 0;
+        }
+
         if (!txObj) {
-          const receiptShowsFailedStatus = await this.checkTxReceiptStatusIsFailed(
-            transactionHash,
-          );
+          meta.pollingAttempts += 1;
 
-          // Case the txObj is evaluated as false, a second check will
-          // determine if the tx failed or it is pending or confirmed
-          if (receiptShowsFailedStatus) {
+          if (pollingAttempts === maxPollingAttempts) {
             const error: Error = new Error(
               'Transaction failed. The transaction was dropped or replaced by a new one',
             );
             this.failTransaction(meta, error);
+            return [meta, false];
           }
+
+          return [meta, true];
```

Comment:
I think the second one is also required because the transaction status is being updated from `submitted` to `confirmed`

```javascript
if (txObj?.blockNumber) {
  meta.status = TransactionStatus.confirmed;
  this.hub.emit(`${meta.id}:confirmed`, meta);
  return [meta, true];
}
```

Reply:
Oh sorry, I misinterpreted which indentation level this was on. Nevermind!